### PR TITLE
Update KMCP/SEARCH: Add nf-test, flip order of input, version bump

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: nf-test.config [hunk 1]
```diff
@@ -13,6 +13,6 @@ config {
 
     // load the necessary plugins
     plugins {
-        load "nft-bam@0.2.0"
-    }
+        load "nft-bam@0.1.1"
```
Comment (jfy133) [inline nf-test.config:16]:
```suggestion
        load "nft-bam@0.2.0"
```

File: nf-test.config [hunk 2]
```diff
@@ -14,5 +14,5 @@ config {
     // load the necessary plugins
     plugins {
         load "nft-bam@0.2.0"
-    }
+        }
```
Comment (jfy133) [inline nf-test.config:17]:
```suggestion
    }
```

File: modules/nf-core/kmcp/search/main.nf [hunk 3]
```diff
@@ -40,7 +40,8 @@ process KMCP_SEARCH {
     def args = task.ext.args ?: ''
     prefix = task.ext.prefix ?: "${meta.id}"
     """
-    touch ${prefix}.gz
+    touch ${prefix}
+    gzip ${prefix}
```
Comment (SPPearce) [inline modules/nf-core/kmcp/search/main.nf:44]: This is how we have generally been doing this
```suggestion
   echo "" | gzip > ${prefix}.gz
```

File: modules/nf-core/kmcp/search/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    tag "kmcp/search"
+    tag "kmcp/compute"
+    tag "kmcp/index"
+
+    setup {
+        run("KMCP_COMPUTE") {
+            script "../../../kmcp/compute/main.nf"
+            process {
+                """
+                input[0] = [[id: 'test'],file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)]
```
Comment (SPPearce) [inline modules/nf-core/kmcp/search/tests/main.nf.test:20]: Can you split this over multiple lines

**Review by SPPearce — APPROVED** [review]

File: modules/nf-core/kmcp/search/main.nf (hunk 3 above)
Comment (jfy133) [reply modules/nf-core/kmcp/search/main.nf:44]: You know what I knew there was something like that but couldn't for the life of me remember it

And I remembered that exactly this morning when I was working on another module with @aidaanva 🤣 thank you!

**Review by SPPearce — APPROVED** [review]

**Pull request merged**